// luojilab-log4js/src/config.js
var SERVER_CONFIG = process.env.SERVER_CONFIG || 'development';

function Configure(logPath) {
  return {
    development: {
      config: {
        "appenders": [
          {
            "type": "console",
            "category": "console"
          },
          {
            "type": "console",
            "category": "out"
          },
          {
            "type": "console",
            "category": "err"
          }
        ],
        "replaceConsole": true,
        "levels": {
          "out": "info",
          "err": "warn",
          "console": "info"
        }
      },
      format: ":method :url :status - :response-time ms"
    },
    testing: {
      config: {
        "appenders": [
          {
            "type": "console",
            "category": "console"
          }, {
            "type": "dateFile",
            "filename": logPath,
            "pattern": "yyyy-MM-dd-hh-out.log",
            "absolute": true,
            "alwaysIncludePattern": true,
            "category": "out"
          }, {
            "type": "dateFile",
            "filename": logPath,
            "pattern": "yyyy-MM-dd-err.log",
            "absolute": true,
            "alwaysIncludePattern": true,
            "category": "err"
          }
        ],
        "replaceConsole": false,
        "levels": {
          "out": "info",
          "err": "warn",
          "console": "info"
        }
      },
      format: ":method :url :status - :response-time ms"
    },
    simulation: {
      config: {
        "appenders": [
          {
            "type": "console",
            "category": "console"
          }, {
            "type": "dateFile",
            "filename": logPath,
            "pattern": "yyyy-MM-dd-hh-out.log",
            "absolute": true,
            "alwaysIncludePattern": true,
            "category": "out"
          }, {
            "type": "dateFile",
            "filename": logPath,
            "pattern": "yyyy-MM-dd-err.log",
            "absolute": true,
            "alwaysIncludePattern": true,
            "category": "err"
          }
        ],
        "replaceConsole": false,
        "levels": {
          "out": "info",
          "err": "warn",
          "console": "info"
        }
      },
      format: ":method :url :status - :response-time ms"
    },
    production: {
      config: {
        "appenders": [
          {
            "type": "console",
            "category": "console"
          }, {
            "type": "dateFile",
            "filename": logPath,
            "pattern": "yyyy-MM-dd-hh-out.log",
            "absolute": true,
            "alwaysIncludePattern": true,
            "category": "out"
          }, {
            "type": "dateFile",
            "filename": logPath,
            "pattern": "yyyy-MM-dd-err.log",
            "absolute": true,
            "alwaysIncludePattern": true,
            "category": "err"
          }
        ],
        "replaceConsole": false,
        "levels": {
          "out": "info",
          "err": "warn",
          "console": "info"
        }
      },
      format: ":remote-addr :http-version :method :url :status - :referrer - :user-agent - :response-time ms"
    }
  }
}

module.exports = function(logPath) {
  return Configure(logPath)[SERVER_CONFIG]
}
